### PR TITLE
feat: add debug server with pprof

### PR DESCRIPTION
**Pull request by enocom**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -434,6 +434,21 @@ To enable Prometheus, use the `--prometheus` flag. This will start an HTTP
 server on localhost with a `/metrics` endpoint. The Prometheus namespace may
 optionally be set with `--prometheus-namespace`.
 
+## Debug server
+
+The Proxy includes support for a debug server on localhost. By default, the
```
Comment (kurtisvg) [inline README.md:439]: > The Proxy includes support for a debug server on localhost

Will this change with the http-addr flag?

File: README.md [hunk 2]
```diff
@@ -434,6 +434,21 @@ To enable Prometheus, use the `--prometheus` flag. This will start an HTTP
...
+debug server is not enabled. To enable the server, pass the `--debug` flag.
+This will start the server on localhost at port 9191. To change the port,
```
Comment (kurtisvg) [inline README.md:441]: Should we reuse the http handler for readiness checks and metrics?

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (enocom) [reply README.md:441]: We'll make this 9091.

Comment (enocom) [reply README.md:441]: We'll change this to `---admin-port`.

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**